CMakeLists.txt

```diff
@@ -83,9 +83,18 @@ add_library(SDL2_net
     SDLnetTCP.c
     SDLnetUDP.c
 )
+
+file(
+  COPY "${CMAKE_CURRENT_SOURCE_DIR}/SDL_net.h"
+  DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/SDL2/"
+)
```

Comment:
I recently moved the sources and headers of the other satellite libraries (sources to `src`, and headers to `include/SDL2`)
SDL2_net was skipped because I assumed it was EOL (which Ryan is denying :wink:)
Let's also do this with SDL2_net, so we don't need to move headers.

With that change, you don't need to configure `SDL2Config.cmake.in` twice and can just add `"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"` and `"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/SDL2>"` as build-time include path.

Reply:
Of course, a complication with this approach is that the configure script also needs some modification.
But in general, [most changes were straightforward](https://github.com/libsdl-org/SDL_image/commit/f7169f3b09f97dbfa316caaf86b7d5acad9df7ed)

Reply:
> With that change, you don't need to configure SDL2Config.cmake.in twice

Configuring `SDL2_netConfig.cmake.in` twice does not really relate to headers since that's handled in the `SDL2_net-*-targets.cmake` files. The main reasons we want to do it twice are:
- Set a proper `PACKAGE_PREFIX_DIR`, which depends on the `INSTALL_DESTINATION` and `INSTALL_PREFIX` provided to `configure_package_config_file`. Arguably, `@PACKAGE_INIT@` could be taken off since neither `set_and_check` or `check_required_components` need to be used, but keep it in mind for SDL3_net 😛.
- Similar to your next comment about guarding the target export when SDL is a subproject, having the "install config" and the "build config" be two separate files would avoid the situation where `SDL2_net-*-targets.cmake` are not exported to the build directory, but there is still a package config that loads no target.

---

Anyway, I moved the files in https://github.com/libsdl-org/SDL_net/pull/80/commits/86ce93d1c105fc9d1a3ffee4054142dde975256d similarly to what you did for the other satellite libraries, let me know what you think!

Reply:
- Can't we just remove `@PACKAGE_INIT@`, or unset it? `PACKAGE_PREFIX_DIR` is intended to be used by the `CMakePackageConfigHelpers` module itself (for paths passed in `PATH_VARS`), which we don't use.
So the variable is effectively a don't care.
- About the case where `SDL2_netConfig.cmake` exists, but no `-*targets.cmake`. How about we do the following in `SDL2_netConfig.cmake`:
    ```cmake
    set(SDL2_net_FOUND FALSE)
    if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/SDL2_net-shared-targets.cmake")
        include("${CMAKE_CURRENT_LIST_DIR}/SDL2_net-shared-targets.cmake")
        set(SDL2_net_FOUND TRUE)
    endif()
    
    if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/SDL2_net-static-targets.cmake")
        include("${CMAKE_CURRENT_LIST_DIR}/SDL2_net-static-targets.cmake")
        set(SDL2_net_FOUND TRUE)
    endif()
    ```
    And also move `export` under the `SDL2NET_INSTALL` conditional. A requirement to configure SDL2_net with `SDL2_INSTALL=ON` makes sense imho.